pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -63,7 +72,23 @@ func (p *Probe) checkConfigStatus() error {
 
 // isEnvoyReady checks to ensure that Envoy is in the LIVE state and workers have started.
 func (p *Probe) isEnvoyReady() error {
-	state, ws, err := util.GetReadinessStats(p.LocalHostAddr, p.AdminPort)
+	// Execute the stats query on Envoy if atleast readinessTTL has expired or
+	// Envoy is not ready at least once.  After Envoy is ready for the first time,
+	// we return cached value to avoid frequent executions of stats query till
+	// cached TTL is reached.
+	if !p.atleastOnceReady || time.Since(p.lastUpdateTime) >= readinessTTL {
```

Comment:
I think if we have atleastOnceReady=true but hit the TTL, then the readiness check fails, we never set atleastOnceReady=false.

Can we add some tests around this?

Reply:
Yes. I am planning to add tests, if we are good with the approach

Reply:
Kuat raised a good question. Do we even need to check readiness after first success? since we don't hot restart now I'm not sure we do?

Reply:
> we never set atleastOnceReady=false.

We do not have to set right?  - because it just indicates if has ever been ready atleast once - used to execute stats check on Envoy every time till it is ready once - just not to delay readiness. And if readiness fails after it is ready once and TTL expired - we correctly return the readiness state.

Added tests. Can you please check if we need to cover more cases?